bin/intermitents.js

```diff
@@ -0,0 +1,112 @@
+const shell = require("shelljs");
+const chalk = require("chalk");
+const path = require("path");
+const _ = require("lodash");
+const minimist = require("minimist");
+const emoji = require("node-emoji");
+
+/**
+ * Checks for intermittents with one of two strategies:
+ * 1. runs each test a number of times
+ * 2. groups tests until failing tests are isolated
+ *
+ * params:
+ *  --group enable test grouping
+ *  --runs <number> set the number of test runs (default: 10)
+ *  --path filters tests given a path
+ */
+
+const args = minimist(process.argv.slice(2), {
+  boolean: ["group"],
+  number: ["runs"],
+  string: ["path"]
+});
+
+const runs = args.runs || 10;
+const testPath = args.path ? `--testPathPattern ${args.path}` : "";
+const child = shell.exec(`jest --listTests ${testPath}`, { silent: true });
+const tests = JSON.parse(child.stdout);
+var log = require("single-line-log").stdout;
+
+const write = msg => process.stdout.write(msg);
+
+function runTest(test) {
+  const file = path.basename(test);
+  log(`${emoji.get("sweat_smile")}  ${file} `);
+
+  const startTime = Date.now();
+  let failed = false;
+  let elapsedTime = 0;
+  let progress = [];
+  _.times(runs).forEach(() => {
+    const out = shell.exec(`jest ${test}`, { silent: true });
+    failed = failed || out.code !== 0;
+    const endTime = Date.now();
+    elapsedTime = Math.round((endTime - startTime) / 1000);
+    if (failed) {
+      progress.push(chalk.red("."));
+    } else {
+      progress.push(".");
+    }
+    const status = failed ? emoji.get("rage") : emoji.get("sweat_smile");
+    const dots = progress.join("");
+    log(`${status}  ${file} ${dots}`);
+  });
+
+  const dots = progress.join("");
+  const status = failed ? emoji.get("rage") : emoji.get("sweat_smile");
```

Comment:
if i am reading this correctly.. this will be set if the last run of the test failed. maybe we want to change it so that the forEach loop has its own failed status, and this uses _.any(test => test.failed) or something?

Reply:
i just realized that we maintain the failed state through the loop.. maybe we could use a for loop and exit early if it fails?

Reply:
i did that earlier but i like having the data of how often it fails... we can always clean it up later